Library/Homebrew/keg_relocate.rb

```diff
@@ -163,6 +163,45 @@ def replace_text_in_files(relocation, files: nil)
     changed_files
   end
 
+  def relocate_build_prefix(keg, old_cellar, new_prefix)
+    # Find binaries which match prefix strings.
+    old_prefix = Pathname(old_cellar).parent.to_s
+    string_matches = []
+    keg.each_unique_file_matching(old_prefix) do |file|
+      string_matches << file
+    end
+    binary_matches = []
+    keg.each_unique_binary_file(old_prefix) do |file|
+      binary_matches << file
+    end
+
+    # Only consider string matches which are binary files with null bytes, and remove any
+    # matches which are sharballs found by text_files.
+    binary_string_matches = (string_matches & binary_matches) - text_files
+
+    # Split binary by null characters into array and substitute new cellar for old cellar.
+    # Null padding is added if the new string is too short.
+    binary_string_matches.each do |binary_file|
+      binary_file.ensure_writable do
+        binary = File.binread binary_file
+        puts "Replacing build prefix in: #{binary_file}"
+        binary_strings = binary.split("\x00")
+        match_indices = binary_strings.each_index.select { |i| binary_strings[i].include?(old_prefix) }
+
+        # Only perform substitution on strings which match prefix regex.
+        match_indices.each do |i|
+          s = binary_strings[i]
+          binary_strings[i] = s.gsub(old_prefix, new_prefix).ljust(s.size, "\x00")
+        end
+
+        # Add back null padding at the end of the binary if needed.
+        patched_binary = binary_strings.join("\x00").ljust(binary.size, "\x00")
+        binary_file.atomic_write patched_binary
+      end
+      apply_ad_hoc_signature(binary_file) if OS.mac?
```

Comment:
Does this need to only be done on ARM?

Reply:
Within `apply_ad_hoc_signature`, there is the line: `return unless Hardware::CPU.arm?` so it seems it will already skip this on Intel.